File: contact-form/myproject/scripts/generate_screenshots.py
import os
import shutil
import subprocess
import time
from playwright.sync_api import sync_playwright

BASE_URL = "http://127.0.0.1:8000"
SCREENSHOT_DIR = "screenshots"
TEMP_FRAME_DIR = "temp_frames"

def ensure_dir(path):
    if not os.path.exists(path):
        os.makedirs(path)

def clear_dir(path):
    if os.path.exists(path):
        shutil.rmtree(path)
    os.makedirs(path)

def set_dark_theme(page):
    page.evaluate("""
        document.documentElement.classList.add('dark');
        localStorage.setItem('app_theme', 'dark');
    """)

def inject_custom_pointer(page):
    page.evaluate("""
    if (!document.getElementById('custom-mouse-style')) {
        const style = document.createElement('style');
        style.id = 'custom-mouse-style';
        style.innerHTML = `
          #custom-mouse-pointer {
            position: fixed;
            top: -100px;
            left: -100px;
            width: 22px;
            height: 22px;
            background: rgba(239, 68, 68, 0.9);
            border: 2px solid #ffffff;
            border-radius: 50%;
            pointer-events: none;
            z-index: 999999;
            transform: translate(-50%, -50%);
            box-shadow: 0 2px 8px rgba(0,0,0,0.5);
          }
          .custom-ripple {
            position: fixed;
            width: 40px;
            height: 40px;
            border: 3px solid rgba(239, 68, 68, 0.9);
            border-radius: 50%;
            pointer-events: none;
            z-index: 999998;
            transform: translate(-50%, -50%) scale(0.2);
            opacity: 1;
            animation: custom-ripple-anim 0.35s ease-out forwards;
          }
          @keyframes custom-ripple-anim {
            to {
              transform: translate(-50%, -50%) scale(1.6);
              opacity: 0;
            }
          }
        `;
        document.head.appendChild(style);
        const pointer = document.createElement('div');
        pointer.id = 'custom-mouse-pointer';
        document.body.appendChild(pointer);
    }
    """)

class AnimationRecorder:
    def __init__(self, page, frames_dir):
        self.page = page
        self.frames_dir = frames_dir
        self.frame_count = 0
        self.current_pos = (400, 300)
        clear_dir(self.frames_dir)

    def capture_frame(self, repeat=1):
        for _ in range(repeat):
            self.frame_count += 1
            filename = os.path.join(self.frames_dir, f"frame_{self.frame_count:04d}.png")
            self.page.screenshot(path=filename)

    def update_pointer(self, x, y):
        self.page.evaluate(f"""
            const p = document.getElementById('custom-mouse-pointer');
            if (p) {{
                p.style.left = '{x}px';
                p.style.top = '{y}px';
            }}
        """)

    def move_to(self, x, y, steps=6):
        start_x, start_y = self.current_pos
        for i in range(1, steps + 1):
            cur_x = start_x + (x - start_x) * (i / steps)
            cur_y = start_y + (y - start_y) * (i / steps)
            self.update_pointer(cur_x, cur_y)
            self.capture_frame(repeat=1)
        self.current_pos = (x, y)

    def move_to_element(self, selector_or_locator, steps=6):
        if isinstance(selector_or_locator, str):
            loc = self.page.locator(selector_or_locator).first
        else:
            loc = selector_or_locator

        if loc.count() == 0:
            return self.current_pos

        box = loc.bounding_box()
        if box:
            target_x = box['x'] + box['width'] / 2
            target_y = box['y'] + box['height'] / 2
            self.move_to(target_x, target_y, steps=steps)
            return target_x, target_y
        return self.current_pos

    def click_element(self, selector_or_locator, steps=6):
        if isinstance(selector_or_locator, str):
            loc = self.page.locator(selector_or_locator).first
        else:
            loc = selector_or_locator

        if loc.count() == 0:
            return

        box = loc.bounding_box()
        if box:
            target_x = box['x'] + box['width'] / 2
            target_y = box['y'] + box['height'] / 2
            self.move_to(target_x, target_y, steps=steps)
            
            # Ripple
            self.page.evaluate(f"""
                const r = document.createElement('div');
                r.className = 'custom-ripple';
                r.style.left = '{target_x}px';
                r.style.top = '{target_y}px';
                document.body.appendChild(r);
                setTimeout(() => r.remove(), 400);
            """)
            self.capture_frame(repeat=2)
            loc.click()
            self.capture_frame(repeat=3)

    def type_text(self, selector_or_locator, text, steps_per_char=1):
        if isinstance(selector_or_locator, str):
            loc = self.page.locator(selector_or_locator).first
        else:
            loc = selector_or_locator

        if loc.count() == 0:
            return

        box = loc.bounding_box()
        if box:
            self.move_to(box['x'] + box['width'] / 2, box['y'] + box['height'] / 2)
            loc.click()
            self.capture_frame(repeat=1)
        
        # Clear existing text if any
        loc.fill("")
        
        # Type character by character
        typed = ""
        for char in text:
            typed += char
            loc.fill(typed)
            loc.dispatch_event("input")
            loc.dispatch_event("change")
            self.capture_frame(repeat=steps_per_char)
        self.capture_frame(repeat=2)


def generate_static_screenshots(playwright):
    print("=== Generating Static Screenshots ===")
    browser = playwright.chromium.launch()
    context = browser.new_context(
        viewport={"width": 1280, "height": 800},
        device_scale_factor=2,
        locale="ja-JP"
    )
    page = context.new_page()

    ensure_dir(SCREENSHOT_DIR)

    # 1. 公開フォーム入力画面 (contact_input.jpg)
    print("Capturing contact_input.jpg...")
    page.goto(f"{BASE_URL}/contact")
    set_dark_theme(page)
    page.fill("#name", "山田 太郎")
    page.fill("#email", "yamada@example.com")
    page.fill("#subject", "商品のお問い合わせ")
    page.fill("#body", "商品の詳細と納期について確認させていただけますでしょうか。\n何卒よろしくお願いいたします。")
    page.evaluate("""
        document.querySelectorAll('input, textarea').forEach(el => {
            el.dispatchEvent(new Event('input', { bubbles: true }));
        });
    """)
    page.wait_for_timeout(300)
    page.screenshot(path=f"{SCREENSHOT_DIR}/contact_input.jpg", type="jpeg", quality=90, full_page=True)

    # 2. 確認画面 (contact_confirm.jpg)
    print("Capturing contact_confirm.jpg...")
    page.click('button[type="submit"]')
    page.wait_for_selector('h2:has-text("お問い合わせ内容確認")')
    set_dark_theme(page)
    page.wait_for_timeout(300)
    page.screenshot(path=f"{SCREENSHOT_DIR}/contact_confirm.jpg", type="jpeg", quality=90, full_page=True)

    # 3. ログイン画面 (admin_login.jpg)
    print("Capturing admin_login.jpg...")
    page.goto(f"{BASE_URL}/login")
    set_dark_theme(page)
    page.fill("#email", "admin@example.test")
    page.wait_for_timeout(300)
    page.screenshot(path=f"{SCREENSHOT_DIR}/admin_login.jpg", type="jpeg", quality=90, full_page=True)

    # 4. 管理画面一覧 (admin_dashboard.jpg)
    print("Capturing admin_dashboard.jpg...")
    page.fill("#password", "local-only-not-a-secret")
    page.click('button[type="submit"]')
    page.wait_for_selector('h2:has-text("お問い合わせ一覧")')
    set_dark_theme(page)
    # 絞り込みアコーディオンを開いた状態
    page.evaluate("""
        const filters = document.querySelector('[x-data="contactFilters"]');
        if (filters && window.Alpine) {
            const data = Alpine.$data(filters);
            if (data && !data.isExpanded) data.isExpanded = true;
        }
    """)
    page.wait_for_timeout(500)
    page.screenshot(path=f"{SCREENSHOT_DIR}/admin_dashboard.jpg", type="jpeg", quality=90, full_page=True)

    # 5. 詳細画面 (admin_show.jpg)
    print("Capturing admin_show.jpg...")
    first_show_link = page.locator('a:has-text("詳細")').first
    first_show_link.click()
    page.wait_for_selector('h2:has-text("お問い合わせ詳細")')
    set_dark_theme(page)
    page.wait_for_timeout(300)
    page.screenshot(path=f"{SCREENSHOT_DIR}/admin_show.jpg", type="jpeg", quality=90, full_page=True)

    browser.close()
    print("Static screenshots generated successfully.")


def generate_demo_public_form_gif(playwright):
    print("=== Generating demo_public_form.gif ===")
    frames_dir = os.path.join(TEMP_FRAME_DIR, "public_form")
    browser = playwright.chromium.launch()
    context = browser.new_context(
        viewport={"width": 1280, "height": 800},
        device_scale_factor=1,
        locale="ja-JP"
    )
    page = context.new_page()
    recorder = AnimationRecorder(page, frames_dir)

    page.goto(f"{BASE_URL}/contact")
    set_dark_theme(page)
    inject_custom_pointer(page)
    recorder.update_pointer(400, 200)
    recorder.capture_frame(repeat=3)

    # 入力
    recorder.type_text("#name", "山田 太郎", steps_per_char=1)
    recorder.type_text("#email", "yamada@example.com", steps_per_char=1)
    recorder.type_text("#subject", "商品の詳細確認について", steps_per_char=1)
    recorder.type_text("#body", "商品の在庫状況と納期の目安について教えいただけますでしょうか。\nよろしくお願いします。", steps_per_char=1)

    recorder.capture_frame(repeat=5)

    # 確認画面へ進む
    recorder.click_element('button[type="submit"]')
    page.wait_for_selector('h2:has-text("お問い合わせ内容確認")')
    set_dark_theme(page)
    inject_custom_pointer(page)
    recorder.update_pointer(600, 300)
    recorder.capture_frame(repeat=5)

    # 送信ボタンクリック
    submit_btn = page.locator('button[name="action"][value="submit"], button[type="submit"]:has-text("送信する")')
    recorder.click_element(submit_btn)
    page.wait_for_selector('h1:has-text("お問い合わせありがとうございました")')
    set_dark_theme(page)
    inject_custom_pointer(page)
    recorder.update_pointer(600, 400)
    recorder.capture_frame(repeat=10)

    browser.close()

    # Convert to GIF via ffmpeg
    gif_path = os.path.join(SCREENSHOT_DIR, "demo_public_form.gif")
    cmd = (
        f'ffmpeg -y -framerate 8 -i "{frames_dir}/frame_%04d.png" '
        f'-vf "scale=800:-1:flags=lanczos,split[s0][s1];[s0]palettegen=max_colors=128[p];[s1][p]paletteuse=dither=bayer" '
        f'"{gif_path}"'
    )
    print("Running ffmpeg for demo_public_form.gif...")
    subprocess.run(cmd, shell=True, check=True)
    print("demo_public_form.gif created successfully.")


def generate_demo_admin_dashboard_gif(playwright):
    print("=== Generating demo_admin_dashboard.gif ===")
    frames_dir = os.path.join(TEMP_FRAME_DIR, "admin_dashboard")
    browser = playwright.chromium.launch()
    context = browser.new_context(
        viewport={"width": 1280, "height": 800},
        device_scale_factor=1,
        locale="ja-JP"
    )
    page = context.new_page()
    recorder = AnimationRecorder(page, frames_dir)

    # 1. ログイン画面
    page.goto(f"{BASE_URL}/login")
    set_dark_theme(page)
    inject_custom_pointer(page)
    recorder.update_pointer(400, 200)
    recorder.capture_frame(repeat=3)

    recorder.type_text("#email", "admin@example.test", steps_per_char=1)
    recorder.type_text("#password", "local-only-not-a-secret", steps_per_char=1)
    recorder.click_element('button[type="submit"]')

    # 2. 一覧画面
    page.wait_for_selector('h2:has-text("お問い合わせ一覧")')
    set_dark_theme(page)
    inject_custom_pointer(page)
    recorder.update_pointer(600, 200)
    recorder.capture_frame(repeat=5)

    # テーマ切替ボタンをクリック（ダーク -> ライト -> ダーク）
    theme_toggle = page.locator('#theme-toggle-button, button[aria-label*="テーマ"], button[aria-label*="theme"], button[x-data*="themeToggle"]').first
    if theme_toggle.count() > 0:
        recorder.click_element(theme_toggle)
        recorder.capture_frame(repeat=5)
        recorder.click_element(theme_toggle)
        set_dark_theme(page)
        recorder.capture_frame(repeat=5)

    # ステータス複数チェックボックス操作
    status_checkboxes = page.locator('input[name="status[]"]')
    if status_checkboxes.count() >= 2:
        recorder.click_element(status_checkboxes.nth(0))
        page.wait_for_timeout(300)
        recorder.capture_frame(repeat=4)
        recorder.click_element(status_checkboxes.nth(1))
        page.wait_for_timeout(300)
        recorder.capture_frame(repeat=4)

    # 検索キー入力
    recorder.type_text("#keyword", "商品", steps_per_char=1)
    page.wait_for_timeout(600)
    recorder.capture_frame(repeat=5)

    # 詳細表示をクリック
    detail_link = page.locator('a:has-text("詳細")').first
    recorder.click_element(detail_link)

    # 3. 詳細画面
    page.wait_for_selector('h2:has-text("お問い合わせ詳細")')
    set_dark_theme(page)
    inject_custom_pointer(page)
    recorder.update_pointer(600, 300)
    recorder.capture_frame(repeat=6)

    # ステータス変更ドロップダウン操作
    status_select = page.locator('select[name="status"]')
    if status_select.count() > 0:
        recorder.move_to_element(status_select)
        status_select.select_option("in_progress")
        recorder.capture_frame(repeat=4)

    # 一覧に戻るリンク
    back_link = page.locator('a:has-text("一覧に戻る")').first
    if back_link.count() > 0:
        recorder.click_element(back_link)
        page.wait_for_selector('h2:has-text("お問い合わせ一覧")')
        set_dark_theme(page)
        inject_custom_pointer(page)
        recorder.capture_frame(repeat=6)

    browser.close()

    # Convert to GIF via ffmpeg
    gif_path = os.path.join(SCREENSHOT_DIR, "demo_admin_dashboard.gif")
    cmd = (
        f'ffmpeg -y -framerate 8 -i "{frames_dir}/frame_%04d.png" '
        f'-vf "scale=800:-1:flags=lanczos,split[s0][s1];[s0]palettegen=max_colors=128[p];[s1][p]paletteuse=dither=bayer" '
        f'"{gif_path}"'
    )
    print("Running ffmpeg for demo_admin_dashboard.gif...")
    subprocess.run(cmd, shell=True, check=True)
    print("demo_admin_dashboard.gif created successfully.")


if __name__ == "__main__":
    with sync_playwright() as playwright:
        generate_static_screenshots(playwright)
        generate_demo_public_form_gif(playwright)
        generate_demo_admin_dashboard_gif(playwright)
    
    # Clean up temp frames
    if os.path.exists(TEMP_FRAME_DIR):
        shutil.rmtree(TEMP_FRAME_DIR)
    print("=== All screenshots and demos successfully updated! ===")

File: todo-app/scripts/generate_screenshots.js
const { chromium } = require('@playwright/test');
const fs = require('fs');
const path = require('path');
const { execSync } = require('child_process');

const BASE_URL = 'http://localhost:8123';
const SCREENSHOT_DIR = path.join(__dirname, '..', 'screenshots');
const TEMP_FRAME_DIR = path.join(__dirname, '..', 'temp_frames');

function ensureDir(dirPath) {
  if (!fs.existsSync(dirPath)) {
    fs.mkdirSync(dirPath, { recursive: true });
  }
}

function clearDir(dirPath) {
  if (fs.existsSync(dirPath)) {
    fs.rmSync(dirPath, { recursive: true, force: true });
  }
  fs.mkdirSync(dirPath, { recursive: true });
}

function injectCustomPointer(page) {
  return page.evaluate(() => {
    if (!document.getElementById('custom-mouse-style')) {
      const style = document.createElement('style');
      style.id = 'custom-mouse-style';
      style.innerHTML = `
        #custom-mouse-pointer {
          position: fixed;
          top: -100px;
          left: -100px;
          width: 22px;
          height: 22px;
          background: rgba(239, 68, 68, 0.9);
          border: 2px solid #ffffff;
          border-radius: 50%;
          pointer-events: none;
          z-index: 999999;
          transform: translate(-50%, -50%);
          box-shadow: 0 2px 8px rgba(0,0,0,0.5);
        }
        .custom-ripple {
          position: fixed;
          width: 40px;
          height: 40px;
          border: 3px solid rgba(239, 68, 68, 0.9);
          border-radius: 50%;
          pointer-events: none;
          z-index: 999998;
          transform: translate(-50%, -50%) scale(0.2);
          opacity: 1;
          animation: custom-ripple-anim 0.35s ease-out forwards;
        }
        @keyframes custom-ripple-anim {
          to {
            transform: translate(-50%, -50%) scale(1.6);
            opacity: 0;
          }
        }
      `;
      document.head.appendChild(style);
      const pointer = document.createElement('div');
      pointer.id = 'custom-mouse-pointer';
      document.body.appendChild(pointer);
    }
  });
}

class AnimationRecorder {
  constructor(page, framesDir) {
    this.page = page;
    this.framesDir = framesDir;
    this.frameCount = 0;
    this.currentPos = { x: 400, y: 300 };
    clearDir(this.framesDir);
  }

  async captureFrame(repeat = 1) {
    for (let i = 0; i < repeat; i++) {
      this.frameCount++;
      const filename = path.join(this.framesDir, `frame_${String(this.frameCount).padStart(4, '0')}.png`);
      await this.page.screenshot({ path: filename });
    }
  }

  async updatePointer(x, y) {
    await this.page.evaluate(({ x, y }) => {
      const p = document.getElementById('custom-mouse-pointer');
      if (p) {
        p.style.left = `${x}px`;
        p.style.top = `${y}px`;
      }
    }, { x, y });
  }

  async moveTo(x, y, steps = 6) {
    const startX = this.currentPos.x;
    const startY = this.currentPos.y;
    for (let i = 1; i <= steps; i++) {
      const curX = startX + (x - startX) * (i / steps);
      const curY = startY + (y - startY) * (i / steps);
      await this.updatePointer(curX, curY);
      await this.captureFrame(1);
    }
    this.currentPos = { x, y };
  }

  async moveToElement(selector, steps = 6) {
    const loc = this.page.locator(selector).first();
    const box = await loc.boundingBox();
    if (box) {
      const targetX = box.x + box.width / 2;
      const targetY = box.y + box.height / 2;
      await this.moveTo(targetX, targetY, steps);
      return { x: targetX, y: targetY };
    }
    return this.currentPos;
  }

  async clickElement(selector, steps = 6) {
    const loc = this.page.locator(selector).first();
    const box = await loc.boundingBox();
    if (box) {
      const targetX = box.x + box.width / 2;
      const targetY = box.y + box.height / 2;
      await this.moveTo(targetX, targetY, steps);

      // Ripple effect
      await this.page.evaluate(({ x, y }) => {
        const r = document.createElement('div');
        r.className = 'custom-ripple';
        r.style.left = `${x}px`;
        r.style.top = `${y}px`;
        document.body.appendChild(r);
        setTimeout(() => r.remove(), 400);
      }, { x: targetX, y: targetY });

      await this.captureFrame(2);
      await loc.click();
      await this.captureFrame(3);
    }
  }

  async typeText(selector, text, stepsPerChar = 1) {
    const loc = this.page.locator(selector).first();
    const box = await loc.boundingBox();
    if (box) {
      await this.moveTo(box.x + box.width / 2, box.y + box.height / 2);
      await loc.click();
      await this.captureFrame(1);
    }

    await loc.fill('');
    let typed = '';
    for (const char of text) {
      typed += char;
      await loc.fill(typed);
      await loc.dispatchEvent('input');
      await loc.dispatchEvent('change');
      await this.captureFrame(stepsPerChar);
    }
    await this.captureFrame(2);
  }
}

async function generateStaticScreenshots(playwright) {
  console.log('=== Generating Static Screenshots ===');
  const browser = await playwright.chromium.launch();
  const context = await browser.newContext({
    viewport: { width: 1280, height: 800 },
    deviceScaleFactor: 2,
    locale: 'ja-JP'
  });
  const page = await context.newPage();

  ensureDir(SCREENSHOT_DIR);

  await page.goto(BASE_URL);
  
  // 今日の日付と明日の日付を生成してダミータスクを注入
  const todayStr = new Date().toISOString().split('T')[0];
  const tomorrowStr = new Date(Date.now() + 86400000).toISOString().split('T')[0];

  await page.evaluate(({ todayStr, tomorrowStr }) => {
    const dummyTodos = [
      {
        id: 'dummy-1',
        title: '報告書の作成と提出',
        detail: '<p>今週の進捗報告書を作成し、マネージャーへメール送信する。<strong>期限厳守！</strong></p>',
        dateType: 'datetime',
        deadline: `${todayStr}T17:00:00`,
        completed: false,
        createdAt: `${todayStr}T09:00:00`
      },
      {
        id: 'dummy-2',
        title: 'スーパーでの買い物',
        detail: '<ul><li>牛乳</li><li>食パン</li><li>卵</li></ul>',
        dateType: 'date',
        deadline: `${tomorrowStr}T00:00:00`,
        completed: false,
        createdAt: `${todayStr}T09:10:00`
      },
      {
        id: 'dummy-3',
        title: 'Gitリポジトリの整理',
        detail: '<p>古いブランチの削除およびドキュメントの整理を行う。</p>',
        dateType: 'none',
        deadline: null,
        completed: true,
        createdAt: `${todayStr}T09:20:00`
      }
    ];
    localStorage.setItem('todo-app-items', JSON.stringify(dummyTodos));
    window.location.reload();
  }, { todayStr, tomorrowStr });
  await page.waitForTimeout(500);

  // 1. メイン画面 (main_screen.jpg)
  console.log('Capturing main_screen.jpg...');
  await page.screenshot({ path: path.join(SCREENSHOT_DIR, 'main_screen.jpg'), type: 'jpeg', quality: 90 });

  // 2. 新規登録モーダル (modal_new.jpg)
  console.log('Capturing modal_new.jpg...');
  await page.click('#btn-new');
  await page.waitForTimeout(300);
  await page.screenshot({ path: path.join(SCREENSHOT_DIR, 'modal_new.jpg'), type: 'jpeg', quality: 90 });
  await page.click('#btn-cancel');
  await page.waitForTimeout(300);

  // 3. 編集モーダル (modal_edit.jpg)
  console.log('Capturing modal_edit.jpg...');
  await page.click('text=報告書の作成と提出');
  await page.waitForTimeout(300);
  await page.screenshot({ path: path.join(SCREENSHOT_DIR, 'modal_edit.jpg'), type: 'jpeg', quality: 90 });
  await page.click('#btn-cancel');

  await browser.close();
  console.log('Static screenshots generated successfully.');
}

async function generateDemoGif(playwright) {
  console.log('=== Generating demo_todo_flow.gif ===');
  const framesDir = path.join(TEMP_FRAME_DIR, 'todo_flow');
  const browser = await playwright.chromium.launch();
  const context = await browser.newContext({
    viewport: { width: 1000, height: 700 },
    deviceScaleFactor: 1,
    locale: 'ja-JP'
  });
  const page = await context.newPage();
  const recorder = new AnimationRecorder(page, framesDir);

  await page.goto(BASE_URL);
  
  await page.evaluate(() => {
    localStorage.clear();
    window.location.reload();
  });
  await page.waitForTimeout(500);

  await injectCustomPointer(page);
  await recorder.updatePointer(400, 200);
  await recorder.captureFrame(5);

  // 1. 新規ボタンをクリック
  await recorder.clickElement('#btn-new');
  await page.waitForTimeout(300);
  await injectCustomPointer(page);
  await recorder.captureFrame(3);

  // 2. タイトルを入力
  await recorder.typeText('#input-title', '牛乳とパンを買う', 1);
  
  // 3. 日時「日のみ」を選択
  await recorder.clickElement('#radio-date-day');
  await recorder.captureFrame(2);

  const tomorrowStr = new Date(Date.now() + 86400000).toISOString().split('T')[0];
  await recorder.clickElement('#dummy-date');
  await page.fill('#input-date', tomorrowStr);
  await page.dispatchEvent('#input-date', 'change');
  await recorder.captureFrame(4);

  // 4. 詳細の入力
  await recorder.moveToElement('.ql-editor');
  await page.click('.ql-editor');
  await page.evaluate(() => {
    const editor = document.querySelector('.ql-editor');
    editor.innerHTML = '<p>近所のスーパーで<strong>特売</strong>の牛乳と食パンを買う。</p>';
  });
  await recorder.captureFrame(10);

  // 5. 保存ボタンをクリック
  await recorder.clickElement('#btn-save');
  await page.waitForTimeout(300);
  await injectCustomPointer(page);
  await recorder.captureFrame(5);

  // 6. 完了トグルをクリック
  await recorder.clickElement('.btn-toggle-complete');
  await recorder.captureFrame(8);

  // 7. 「完了済」タブに切り替え
  await recorder.clickElement('#tab-completed');
  await recorder.captureFrame(6);

  // 8. 「すべて」タブに戻す
  await recorder.clickElement('#tab-all');
  await recorder.captureFrame(4);

  // 9. 完済を削除ボタンをクリック
  page.once('dialog', async dialog => {
    console.log(`Dialog message: ${dialog.message()}`);
    await dialog.accept();
  });
  await recorder.clickElement('#btn-delete-completed');
  await page.waitForTimeout(500);
  await injectCustomPointer(page);
  await recorder.captureFrame(10);

  await browser.close();

  // Convert to GIF via ffmpeg
  const gifPath = path.join(SCREENSHOT_DIR, 'demo_todo_flow.gif');
  const cmd = `ffmpeg -y -framerate 10 -i "${framesDir}/frame_%04d.png" -vf "scale=800:-1:flags=lanczos,split[s0][s1];[s0]palettegen=max_colors=128[p];[s1][p]paletteuse=dither=bayer" "${gifPath}"`;
  console.log('Running ffmpeg for demo_todo_flow.gif...');
  execSync(cmd);
  console.log('demo_todo_flow.gif created successfully.');
}

async function main() {
  const playwright = require('@playwright/test');
  await generateStaticScreenshots(playwright);
  await generateDemoGif(playwright);
  
  if (fs.existsSync(TEMP_FRAME_DIR)) {
    fs.rmSync(TEMP_FRAME_DIR, { recursive: true, force: true });
  }
  console.log('=== All screenshots and demos successfully updated! ===');
}

main().catch(err => {
  console.error(err);
  process.exit(1);
});
